plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+	"github.com/stellar/kelp/support/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func connectTestDb() *sql.DB {
+	postgresDbConfig := &postgresdb.Config{
+		Host:      "localhost",
+		Port:      5432,
+		DbName:    "test_database",
+		User:      os.Getenv("POSTGRES_USER"),
+		SSLEnable: false,
+	}
+
+	_, e := postgresdb.CreateDatabaseIfNotExists(postgresDbConfig)
+	if e != nil {
+		panic(e)
+	}
+
+	db, e := sql.Open("postgres", postgresDbConfig.MakeConnectString())
+	if e != nil {
+		panic(e)
+	}
+	return db
+}
+
+var testNativeAsset hProtocol.Asset = hProtocol.Asset{Type: "native"}
+
+func TestMakeFilterVolume(t *testing.T) {
+	testAssetDisplayFn := model.AssetDisplayFn(func(asset model.Asset) (string, error) {
+		sdexAssetMap := map[model.Asset]hProtocol.Asset{
+			model.Asset("XLM"): testNativeAsset,
+		}
+		assetString, ok := sdexAssetMap[asset]
+		if !ok {
+			return "", fmt.Errorf("cannot recognize the asset %s", string(asset))
+		}
+		return utils.Asset2String(assetString), nil
+	})
+
+	testEmptyVolumeConfig := &VolumeFilterConfig{}
+	db := connectTestDb()
+
+	emptyConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, []string{})
+	if e != nil {
+		t.Log("could not make empty config volume query")
+		return
+	}
+
+	marketIDs := []string{"marketID"}
+	marketConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, marketIDs)
+	if e != nil {
+		t.Log("could not make market config volume query")
+		return
+	}
+
+	accountIDs := []string{"accountID"}
+	accountConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", accountIDs, []string{})
+	if e != nil {
+		t.Log("could not make account config volume query")
+		return
+	}
+
+	accountMarketConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", accountIDs, marketIDs)
+	if e != nil {
+		t.Log("could not make account and market config volume query")
+		return
+	}
+
+	testCases := []struct {
+		name           string
+		configValue    string
+		exchangeName   string
+		tradingPair    *model.TradingPair
+		assetDisplayFn model.AssetDisplayFn
+		baseAsset      hProtocol.Asset
+		quoteAsset     hProtocol.Asset
+		db             *sql.DB
+		config         *VolumeFilterConfig
+
+		wantError        error
+		wantSubmitFilter SubmitFilter
+	}{
+		// Note: config can be additional market IDs, make sure to test additionalMarketIds and optionalAccountIDs
+		{
+			name:             "failure - display base",
+			configValue:      "",
+			exchangeName:     "",
+			tradingPair:      &model.TradingPair{Base: "FAIL", Quote: ""},
+			assetDisplayFn:   testAssetDisplayFn,
+			baseAsset:        hProtocol.Asset{},
+			quoteAsset:       hProtocol.Asset{},
+			db:               db,
+			config:           &VolumeFilterConfig{},
+			wantError:        fmt.Errorf("could not convert base asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
```

Comment:
a case like this is more appropriate in the tests for asset display function, better to remove here